src/buffer.cpp

```diff
@@ -263,7 +263,7 @@ bool CNetBuf::Put ( const CVector<uint8_t>& vecbyData, int iInSize )
         // copy new data in internal buffer
         const int iNumBlocks = iInSize / iBlockSize;
 
-        for ( int iBlock = 0; iBlock < iNumBlocks; iBlock++ )
+        for ( long iBlock = 0; iBlock < iNumBlocks; iBlock++ )
```

Comment:
```suggestion
        // iBlock is a long to avoid overflowing a mulitplication later in the code
        for ( long iBlock = 0; iBlock < iNumBlocks; iBlock++ )
```